### PR TITLE
adds email verification changes in new JWT impl.

### PR DESCRIPTION
**Pull request by ameeakshay**

### CODE REVIEW

File: app/routes/routes.js [hunk 1]
```diff
@@ -83,7 +84,22 @@ module.exports = (app, models) => {
...
                 else {
-                    return res.json({token: jwt.sign({ email: user.email, id: user.id, type: req.body.type }, 'ClientVendor')});
+                
+                        var Verify = models.verification;
+
+                        Verify.findOne({where: {id : user.id}}).then(function(verify) {
+                        
+                                temp.status = 200;
+                                console.log(verify);
+                                if (!verify.emailverified) {
```
Comment (ArihantRJain) [inline app/routes/routes.js:94]: Can we use 403 as the status code for Not Verified User?

File: app/models/verification.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
+module.exports = function(sequelize, Sequelize) {
```
Comment (ArihantRJain) [inline app/models/verification.js:2]: Email Verification is applicable for Clients only. So, can we add a Foreign Key Constraint for the Verification Model?

File: server.js [hunk 3]
```diff
@@ -63,6 +65,18 @@ app.use(function(req, res, next) {
...
 });
 
+mailer.extend(app, {
+  from: 'Venderapp',
+  host: 'smtp.gmail.com', // hostname
+  secureConnection: true, // use SSL
+  port: 465, // port for secure SMTP
+  transportMethod: 'SMTP', // default is SMTP. Accepts anything that nodemailer accepts
+  auth: {
+  user: 'radienerga@gmail.com', // gmail id
```
Comment (ArihantRJain) [inline server.js:75]: Are we using radienerga@gmail.com to send the Verification Mails?
After integrating with AWS servers, will we still use the express-mailer?

Comment (ameeakshay) [reply server.js:75]: As of now yes,....
I am not sue if we have a separate mailing server in AWS.

File: app/routes/routes.js (hunk 1 above)
Comment (ameeakshay) [reply app/routes/routes.js:94]: Yes we can add that

File: app/models/verification.js (hunk 2 above)
Comment (ameeakshay) [reply app/models/verification.js:2]: Yes, but for vendor verification we would still need a factor to verify them once it gets verified offline, so at that time we can use this same module to verify the vendors also

**Review by ArihantRJain — CHANGES_REQUESTED** [review]
Update the status code for Not Verified User to 403. 

We can look at using the same model for Vendor Verification later.

File: app/models/verification.js (hunk 2 above)
Comment (ameeakshay) [reply app/models/verification.js:2]: We can add the mailing option only for clients, that can be done, but I feel we would still need the verification module for vendors.

**Pull request merged**